SharedCode/src/commonMain/kotlin/http/RequestLock.kt

```diff
@@ -0,0 +1,33 @@
+package com.jetbrains.handson.mpp.mobile.http
+
+import com.soywiz.klock.DateTime
+import kotlin.random.Random
+
+class RequestLock {
+    class RequestKey(val id: String)
```

Comment:
Do we need this RequestKey class? Can we not just use a string?

I also think this could be condensed significantly if we do that, e.g.:

```kotlin
class RequestLock {

    private var rng = Random(DateTime.nowUnixLong())

    suspend fun <T> attempt(func: suspend () -> T): T? {
        val key = "${DateTime.nowUnixLong()}::${rng.nextInt()}"
        val result = func()
        return if (key == this.key) result else null
    }
}
```

(I haven't checked this works, but that sort of thing)